### PR TITLE
Updating user's information

### PR DESCRIPTION
**Pull request by txiong4**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -27,5 +29,39 @@ app.get("/register", (req, res) => {
 app.get("/login", (req, res) => {
   res.render("login");
 });
+app.put("/:type", async(req, res, next) => {
+  if(req.params.type === "student")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
+    const type = token.type;
```
Comment (jimmoua) [inline app.js]: What if verification fails? This causes `token.type` to be invalid. You should only search the token if the verification passes.

File: app.js [hunk 2]
```diff
@@ -27,5 +29,39 @@ app.get("/register", (req, res) => {
...
+      return res.sendStatus(401);
+    await updateParentInfo(token.user, req.body.firstName, req.body.lastName);
+  }
+  else if(req.params.type === "tutor")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
+    const type = token.type;
+    if(token === false && type !== "tutor")
+      return res.sendStatus(401);
+    await updateTutorInfo(token.user, req.body.firstName, req.body.lastName, req.body.bio, req.body.gender);
```
Comment (jimmoua) [inline app.js:47]: This function can return `false` to denote that it failed. In that case, send in a `500`.

File: tests/app.put.test.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+const app = require("../app");
```
Comment (jimmoua) [inline tests/app.put.test.js]: You should probably rename the file to be better. `updateUserInformationEndpoint.test.js` would be better.

File: tests/app.put.test.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+beforeAll(async() =>{
+  await createTables();
+});
+afterAll(async() =>{
+  await wipeDBTables();
+});
+
+//Student
+describe("PUT/student", () =>{
```
Comment (jimmoua) [inline tests/app.put.test.js]: `describe("PUT /student`

File: tests/app.put.test.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    expect(findStudent.first_name).toBe("Tim");
+    expect(findStudent.last_name).toBe("JOHN");
+    expect(findStudent.bio).toBe("Goodbye Lads");
+    expect(findStudent.gender).toBe("F");
+  
+  });
+});
+
+//Parent
+describe("PUT/parent", () =>{
```
Comment (jimmoua) [inline tests/app.put.test.js]: `"PUT /parent"`

File: app.js [hunk 6]
```diff
@@ -27,5 +29,39 @@ app.get("/register", (req, res) => {
...
+    const type = token.type;
+    if(token === false && type !== "student")
+    {
+      return res.sendStatus(401);
+    }
+    await updateStudentInfo(token.user, req.body.firstName, req.body.lastName, req.body.bio, req.body.gender);
+  }
+  else if(req.params.type === "parent")
+  {
+    const token = await jwtVerify(req.signedCookies.user);
```
Comment (jimmoua) [inline app.js]: This is repeated many tokens. Avoid repeating yourself in code. Verify the token and then you can extract `type` from it.

**Review by jimmoua — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (txiong4) [reply app.js]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

File: app.js (hunk 2 above)
Comment (txiong4) [reply app.js:47]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

File: tests/app.put.test.js (hunk 3 above)
Comment (txiong4) [reply tests/app.put.test.js]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

File: tests/app.put.test.js (hunk 4 above)
Comment (txiong4) [reply tests/app.put.test.js]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

File: tests/app.put.test.js (hunk 5 above)
Comment (txiong4) [reply tests/app.put.test.js]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

File: app.js (hunk 6 above)
Comment (txiong4) [reply app.js]: 9a4a464832b8a05a2ec4f1893f42397a3c4a522b

**Review by jimmoua — APPROVED** [review]

**Pull request merged**